R/annotdata.R

```diff
@@ -103,9 +98,49 @@ addDataLabelAnnotations <- function(p, type, name, data.label.xpos, data.label.y
               textposition = textalign, showlegend = FALSE, hoverinfo = "skip",
               legendgroup = if (is.stacked) "all" else i)
     return(p)
-
 }
 
+
+# Returns data labels after annotations are applied
+# The 'customPoints' attribute is a component of the ChartLabels attribute for powerpoint
+# They are assumed to be attached to the data labels, and an updated version is
+# re-attached to the data labels returned by this function.
+# This function can be called iteratively, but clean.pt.segs should only TRUE on the last call
+# Note that the full column of data should always be supplied because
+# strong assumptions are mode about the indexing
+
+applyAllAnnotationsToDataLabels <- function(data.label.text, annotation.list, 
+    annot.data, series.index, rows.to.show,
+    chart.type, clean.pt.segs = FALSE)
```

Comment:
Documentation is given in lines 135-144 of d723362. I don't think the customPoints part can be particularly obvious for anyone, but there are references to the relevant functions described in the comment.

Reply:
In 8e51af8, I've moved applyAllAnnotationsToDataLabels to sit just above getPointSegmentsForPPT so that all the functions related to updating the ChartLabels attribute for PPT exporting are grouped together.
I've added some more comments for this function because for most of the charting functions it is the top-level function for updating ChartLabels (with the exception Radar which internally uses a different data structure).

The comments don't use the @param structure because its hard to explain each of them separately. But collectively, the description covers all 7 of the parameters